File: Benchmarks/enterprise/python/django/testcode/benchmark_test_27802.py
# SPDX-License-Identifier: Apache-2.0
from django.http import JsonResponse
import logging
import ast


def BenchmarkTest27802(request):
    query_array = request.GET.getlist('items')[0] if request.GET.getlist('items') else ''
    try:
        data = str(ast.literal_eval(query_array))
    except (ValueError, SyntaxError):
        data = query_array
    logging.info('User action: ' + str(data))
    return JsonResponse({"saved": True})

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest27802.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest27802 {

    private static final java.util.concurrent.atomic.AtomicReference<String> atomicValue = new java.util.concurrent.atomic.AtomicReference<>();
    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(BenchmarkTest27802.class);

    @PostMapping("/BenchmarkTest27802")
    public void BenchmarkTest27802(@RequestParam("field") String field, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String fieldValue = field != null ? field : "";
        atomicValue.set(fieldValue);
        String data = atomicValue.get();
        LOG.info("audit actor={} action=revoke_sessions target={}", request.getSession().getAttribute("user"), data);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
